Sources/TuistKit/Services/AuthTokenRefreshService.swift

```diff
@@ -0,0 +1,97 @@
+import FileSystem
+import Foundation
+import Mockable
+import Path
+import ServiceContextModule
+import TuistCore
+import TuistLoader
+import TuistServer
+import TuistSupport
+
+@Mockable
+protocol AuthTokenRefreshServicing: AnyObject {
+    func run(
+        directory: String?
+    ) async throws
+}
+
+final class AuthTokenRefreshService: AuthTokenRefreshServicing {
```

Comment:
```suggestion
struct AuthTokenRefreshService: AuthTokenRefreshServicing {
```

let's make it a struct